app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,44 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DiffUtil;
+
+import java.util.Objects;
+
+import ru.yandex.practicum.contacts.presentation.main.ContactUi;
+
+// public class BaseListDiffCallback<D> extends DiffUtil.ItemCallback<D> implements ListDiffInterface<BaseListDiffCallback>
+public class BaseListDiffCallback <D extends ListDiffInterface<D>> extends DiffUtil.ItemCallback<D> {
+
+
+    public boolean areItemsTheSame(@NonNull D oldItem, @NonNull D newItem) {
+        return oldItem.hashCode() == newItem.hashCode();
```

Comment:
```suggestion
        return oldItem.theSameAs(newItem);
```
Здесь нужно так написать проверку